### PR TITLE
Docs/sector model api

### PR DESCRIPTION
**Pull request by willu47**
Still a few more bits to add to this - it's largely a show and tell on how to use the data handler from within the SectorModel class.  

- @tomalrussell could you check through my examples (e.g. am I using the enum correctly?)

I'm ticking off what I think is the most common workflow that will need to be implemented in the simulate method:
- [x] Retrieve model input and parameter data from the data handler and unpacking the numpy array with reference to the interval and regional definitions
- [x] Write or pass this data to the wrapped model - database, text file, pass into python, pass into command line
- [x] Run the model - python, from command line
- [x] Retrieve results from the model
- [x] Write results back to the data handler - building a numpy array in the correct shape with reference to the interval and regional definitions

### CODE REVIEW

File: docs/getting_started.rst [hunk 1]
```diff
@@ -509,33 +521,284 @@ is required:
...
+
+To access model input data from the timestep prior to the current timestep, 
+you can use the following argument:
+
+.. literalinclude:: ../smif/sample_project/models/energy_demand.py
+   :language: python
+   :lines:  42-44
+   :dedent: 8
+
+Writing data to a database
```
Comment (tomalrussell) [inline docs/getting_started.rst]: Suggest reordering these sections from simple->more involved. Possibly: python, command-line, text, database.

File: smif/sample_project/models/energy_demand.py [hunk 2]
```diff
@@ -11,27 +12,62 @@ def initialise(self, initial_conditions):
...
+                                                       data.timesteps)
+        self.logger.info("Base year energy demand in %s was %s", base_year,
+                         base_energy_demand)
+
+        # Demonstrates how to get the value for a model input from the previous
+        # timeperiod
+        if now > base_year:
+            prev_year_enum = RelativeTimestep.PREVIOUS
+            prev_energy_demand = data.get_data('energy_demand', prev_year_enum)
+            prev_year = base_year_enum.resolve_relative_to(now,
```
Comment (tomalrussell) [inline smif/sample_project/models/energy_demand.py]: Should be `prev_year = prev_year_enum.resolve_relative_to(now,`

File: smif/sample_project/models/energy_demand.py [hunk 3]
```diff
@@ -11,27 +12,62 @@ def initialise(self, initial_conditions):
...
+
+        # Demonstrates how to get the value for a model input
+        # (defaults to the current time period)
+        current_energy_demand = data.get_data('energy_demand')
+        self.logger.info("Current energy demand in %s is %s",
+                         now, current_energy_demand)
+
+        # Demonstrates how to get the value for a model input from the base
+        # timeperiod
+        base_year_enum = RelativeTimestep.BASE
```
Comment (tomalrussell) [inline smif/sample_project/models/energy_demand.py]: Suggest not bothering with the local variable `base_year_enum` here (and `prev_...` below) and use `RelativeTimestep.BASE` directly.

Comment (tomalrussell) [reply smif/sample_project/models/energy_demand.py]: Does this naming feel ergonomic? Could do `Timestep.BASE_YEAR`, `data.get_base_year()`, `data.base_timestep` (to match `data.current_timestep`) ...

File: smif/sample_project/models/energy_demand.py [hunk 4]
```diff
@@ -11,27 +12,62 @@ def initialise(self, initial_conditions):
...
+                                                              data.timesteps)
+        self.logger.info("Base year energy demand in %s was %s", base_year,
+                         base_energy_demand)
+
+        # Demonstrates how to get the value for a model input from the previous
+        # timeperiod
+        if now > base_year:
+            prev_energy_demand = data.get_data('energy_demand',
+                                               RelativeTimestep.PREVIOUS)
+            prev_year = RelativeTimestep.PREVIOUS.resolve_relative_to(
```
Comment (willu47) [inline smif/sample_project/models/energy_demand.py]: Getting the value of a previous year or the base year could be streamlined - perhaps adding a helper method to the data_handle?

Comment (tomalrussell) [reply smif/sample_project/models/energy_demand.py]: @willu47 how do the helper properties/methods below look?

**Comment by tomalrussell** [comment]
Apologies - I rebased your branch on autopilot. All yours now - let me know if the helper methods and properties look good.

**Pull request merged**